cads_adaptors/adaptors/cds.py

```diff
@@ -84,29 +84,60 @@ def _pre_retrieve(self, request: Request, default_download_format="zip"):
 
     def make_download_object(
         self,
-        paths: Union[str, list],
+        paths: str | list[str],
         **kwargs,
     ):
         from cads_adaptors.tools import download_tools
 
-        # Allow possibility of over-riding the download format from the adaptor
-        download_format = kwargs.get("download_format", self.download_format)
-
+        # Ensre paths and filenames are lists
         paths = ensure_list(paths)
         filenames = [os.path.basename(path) for path in paths]
         # TODO: use request-id instead of hash
         kwargs.setdefault(
             "base_target", f"{self.collection_id}-{hash(tuple(self.input_request))}"
         )
 
+        # Allow possibility of over-riding the download format from the adaptor
+        download_format = kwargs.get("download_format", self.download_format)
+        if isinstance(download_format, (list, tuple)):
+            download_format = download_format[0]
+        elif isinstance(download_format, set):
+            download_format = list(download_format)[0]
```

Comment:
You could also wrap the list or tuple with `list`:
```suggestion
        download_format = list(download_format)[0]
```